File: agent/library/src/main/java/org/matilda/di/MatildaConnectionModule.java
package org.matilda.di;

import dagger.Module;
import dagger.Provides;
import org.matilda.MatildaConnection;

import java.io.InputStream;
import java.io.OutputStream;

@Module
public class MatildaConnectionModule {
    private final MatildaConnection mMatildaConnection;

    public MatildaConnectionModule(MatildaConnection matildaConnection) {
        mMatildaConnection = matildaConnection;
    }

    @Provides
    MatildaConnection connection() {
        return mMatildaConnection;
    }

    @Provides
    InputStream inputStream() {
        return mMatildaConnection.inputStream;
    }

    @Provides
    OutputStream outputStream() {
        return mMatildaConnection.outputStream;
    }
}

File: agent/library/src/main/java/org/matilda/di/MatildaConnectionModule.kt
package org.matilda.di

import dagger.Module
import dagger.Provides
import org.matilda.MatildaConnection
import java.io.InputStream
import java.io.OutputStream

@Module
class MatildaConnectionModule(private val mMatildaConnection: MatildaConnection) {
    @Provides
    fun connection(): MatildaConnection {
        return mMatildaConnection
    }

    @Provides
    fun inputStream(): InputStream {
        return mMatildaConnection.inputStream
    }

    @Provides
    fun outputStream(): OutputStream {
        return mMatildaConnection.outputStream
    }
}
